testsuite/features/support/commonlib.rb

```diff
@@ -18,17 +18,20 @@ def generate_temp_file(name, content)
 end
 
 # extract terminals from a retail yaml configuration
-def get_terminals_from_yaml(name)
+def get_terminals_from_yaml
+  name = File.dirname(__FILE__) + '/../upload_files/massive-import-terminals.yml'
```

Comment:
I would prefer to have that path not repeated in more than one place, so easier to maintain.
I suggest to move it to testsuite/features/support/constants.rb

Reply:
It's not possible as it's dynamically computed from path to `commonlib.rb`.
Additionally, if we move the testsuite to another path on the controller, it would fail.

Reply:
That path was only set once before that PR. That's what I would like to keep. Where exactly I don't really care.

Reply:
> That path was only set once before that PR. That's what I would like to keep. Where exactly I don't really care.

 @srbarrios That's factually wrong, it was set in about one dozen scenarios.

> What I would do in your case, is to have a constant for /../upload_files to use everywhere we need and another for massive-import-terminals.yml which is the file for those example tests, both in constants.rb

We can do that, indeed. Or only the `massive-import-terminals.yml` part. In that case the code would read:

``` name = File.dirname(__FILE__) + '/../upload_files/' + massive_import_file```

Putting '/../upload_files/' in the constants is dangerous as this is **relative to current ruby file**. Depending in which ruby file you would use it, it would work or not.

To me, it's plain useless, as this file name is something we are not going to change (remember YAGNI ?), and even if we change it, a quick `grep` should do the trick. But I won't restart this discussion again...

What I'm saying is that we are currently not doing it for any of the other files in `upload_files`. So if we want to do it, we should do it for all of them, not only one.